src/NServiceBus.Core/Transports/OutgoingMessage.cs

```diff
@@ -0,0 +1,32 @@
+namespace NServiceBus.Transports
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The message going out to the transport
+    /// </summary>
+    public class OutgoingMessage
+    {
+        /// <summary>
+        /// Constructs the message
+        /// </summary>
+        /// <param name="headers">The headers associated with this message</param>
+        /// <param name="body">The body of the message</param>
+        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
```

Comment:
But there are still benefits, it all depends how the serialization is
performed, if we allocate a byte[] and that is 80kb+ it will go into LOH ,
on the other hand the stream won't I the serialization is smart enough to
batch the allocations in small chunks.


On Tuesday, March 10, 2015, Daniel Marbach <notifications@github.com> wrote:

> In src/NServiceBus.Core/Transports/OutgoingMessage.cs
> <https://github.com/Particular/NServiceBus/pull/2678#discussion_r26102516>
> :
>
> > @@ -0,0 +1,32 @@
> > +namespace NServiceBus.Transports
> > +{
> > +    using System.Collections.Generic;
> > +
> > +    /// <summary>
> > +    /// The message going out to the transport
> > +    /// </summary>
> > +    public class OutgoingMessage
> > +    {
> > +        /// <summary>
> > +        /// Constructs the message
> > +        /// </summary>
> > +        /// <param name="headers">The headers associated with this message</param>
> > +        /// <param name="body">The body of the message</param>
> > +        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
>
> Agree it was difficult for me to see the benefit of stream in the outgoing
> message. But not in general. In the receiver side it can be beneficial. The
> question is: Are there any transports which support streaming of outgoing
> messages?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2678/files#r26102516>.
>